### ISSUE TITLE
Cancel request exception

### ISSUE THREAD

**Issue by simplezhli** [flutterchina/dio#1089]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |   
| ------------------------------ | ----------------------------------------------------- | 
| Platform Name                  |  android                          |    
| Platform Version               | android 10.0                              |     
| Flutter version              | 2.0.2                           |     
| Dio Version                    | 4.0.0-prev2                                  |    
| Android Studio  | Android Studio 4.1.2              |     
| Repro rate                     | all the time (100%)    |   

### Issue Description and Steps

i use this example :

```dart

CancelToken _cancelToken = CancelToken();
if (!_cancelToken.isCancelled) {
  _cancelToken.cancel();
}

```
 I get the following error :
```
I/flutter (25698): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (25698): The following LateError was thrown while finalizing the widget tree:
I/flutter (25698): LateInitializationError: Field 'requestOptions' has not been initialized.
I/flutter (25698): 
I/flutter (25698): When the exception was thrown, this was the stack:
I/flutter (25698): #0      CancelToken.requestOptions (package:dio/src/cancel_token.dart)
I/flutter (25698): #1      CancelToken.cancel (package:dio/src/cancel_token.dart:40:23)
......
```

https://github.com/flutterchina/dio/blob/7534c1c9aa4577914a176e055a915b9b1220027a/dio/lib/src/cancel_token.dart#L27-L44

**Comment by llfbandit** [comment]
I can reproduce this too, when quickly swiping between page views for example.
Request options were optional in v3, and seems still fine from application point of view and interceptors.

**Comment by wendux** [comment]
fixed in 4.0.0-prev3

**Comment by simplezhli** [comment]
Thank you, I updated to 4.0.0-prev3 , it can work.
